oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/NimbusReactiveJwtDecoder.java

```diff
@@ -177,10 +181,13 @@ private Jwt createJwt(JWT parsedJwt, JWTClaimsSet jwtClaimsSet) {
 
 	private Jwt validateJwt(Jwt jwt) {
 		OAuth2TokenValidatorResult result = this.jwtValidator.validate(jwt);
-
 		if ( result.hasErrors() ) {
-			String message = result.getErrors().iterator().next().getDescription();
-			throw new JwtValidationException(message, result.getErrors());
+			final Collection<OAuth2Error> errors = result.getErrors();
+			final Optional<String> validationErrorString = errors.stream()
```

Comment:
I agree that this more thorough check is an improvement. For now, using `Optional` and `Stream` are too taxing on the garbage collector to be used in Spring Security. 

Will you please change this to a for loop that finds the first non-null description? You could probably do that in a separate method.